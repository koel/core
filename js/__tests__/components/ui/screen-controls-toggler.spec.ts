import each from 'jest-each'
import Component from '@/components/ui/screen-controls-toggler.vue'
import isMobile from 'ismobilejs'
import { shallow } from '@/__tests__/adapter'

describe('components/song/list-controls-toggler', () => {
  beforeEach(() => (isMobile.phone = true))
  afterEach(() => (isMobile.phone = false))

  it('renders properly', () => {
    expect(shallow(Component)).toMatchSnapshot()
  })

  each([[true], [false]]).test('shows/hides properly', shouldShow => {
    const wrapper = shallow(Component, { propsData: { showingControls: shouldShow } })
    expect(wrapper.has('.toggler.fa-angle-up')).toBe(shouldShow)
  })

  it('emits event', () => {
    const wrapper = shallow(Component)
    wrapper.click('.controls-toggler')
    expect(wrapper.hasEmitted('toggleControls')).toBe(true)
  })
})
