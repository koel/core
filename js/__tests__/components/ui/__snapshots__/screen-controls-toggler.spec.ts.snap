// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/song/list-controls-toggler renders properly 1`] = `<span class="controls-toggler text-orange"><i class="fa fa-angle-up toggler"></i></span>`;
